### PR TITLE
Added tests for Function::Parameters parser

### PR DESCRIPTION
**Pull request by jimmygoogle**
I added some tests for the Perl Function::Parameters parser

### CODE REVIEW

File: Units/function-parameters.r/c-fp-test.d/expected.tags [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fun	input.c	/^int fun(void) $/;"	f	typeref:typename:int
+method	input.c	/^int method(void) $/;"	f	typeref:typename:int
```
Comment (masatake) [inline Units/function-parameters.r/c-fp-test.d/expected.tags:2]: A whitespace after `(void)` is not needed.

File: Units/function-parameters.r/c-fp-test.d/expected.tags [hunk 2]
```diff
@@ -0,0 +1,3 @@
+fun	input.c	/^int fun(void) $/;"	f	typeref:typename:int
```
Comment (masatake) [inline Units/function-parameters.r/c-fp-test.d/expected.tags:1]: A whitespace after `(void)` is not needed.

File: Units/function-parameters.r/c-fp-test.d/expected.tags [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+method	input.c	/^int method(void) $/;"	f	typeref:typename:int
+sub	input.c	/^int sub(void) $/;"	f	typeref:typename:int
```
Comment (masatake) [inline Units/function-parameters.r/c-fp-test.d/expected.tags:3]: A whitespace after `(void)` is not needed.

File: Units/function-parameters.r/c-fp-test.d/input.c [hunk 4]
```diff
@@ -0,0 +1,14 @@
+int method(void)
```
Comment (masatake) [inline Units/function-parameters.r/c-fp-test.d/input.c:1]: ...becase there is no whitespace after `(void)`.

File: Units/function-parameters.r/python-fp-test.d/expected.tags [hunk 5]
```diff
@@ -0,0 +1,3 @@
+fun	input.py	/^def fun():$/;"	f
```
Comment (masatake) [inline Units/function-parameters.r/python-fp-test.d/expected.tags:1]: ctags never runs Perl parser ( and FunctionParameter subparser ) for .py files.
So this test case doesn't make sense as a test case for FunctionParameter subparser.
This test case tests the main part of ctags.
Please remove this test case.

**Review by masatake — CHANGES_REQUESTED** [review]
Thank you.

**Comment by masatake** [comment]
Thank you!

**Comment by jimmygoogle** [comment]
Thanks for your help with the parser. I wish I could have done it but I
dont know C that well.

**Comment by masatake** [comment]
I got a question. c-fp-test.d/input.c is not for testing FunctionParameters as I wrote python-fp-test.d, isn't it?

**Comment by jimmygoogle** [comment]
Well I added tests for C and Python to prove the parser didn't try go
beyond the Perl files.

**Comment by masatake** [comment]
> Well I added tests for C and Python to prove the parser didn't try go beyond the Perl files.

I deleted them in #2870 there is no code choosing a parser in the FunctionParameters subparser and Perl parser.
The testing cases for choosing a parser for an input file are at Tmain directory.

**Comment by jimmygoogle** [comment]
OK I see, I had just added to make sure the parser didnt pick them up.

**Pull request merged**